# Graded assignment 1

Noah Woerdman 2009606


# Packages

We need the package `tidyverse`

In [27]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [28]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.0962233 0.5581773 0.4389170 0.5077161 0.3410162 0.9828684 0.7063468
2 0.2503401 0.1780675 0.1040479 0.8954333 0.5061819 0.4037906 0.2082980
  V8        V9        V10       V11       V12       V13       V14      
1 0.4958303 0.4528383 0.4760338 0.9549970 0.3445157 0.3476296 0.4209120
2 0.1315735 0.3748156 0.2358907 0.2915737 0.5953778 0.3505011 0.7620304
  V15        V16       V17       V18       V19       V20      
1 0.01226698 0.7141661 0.2113478 0.2014453 0.8727178 0.5550363
2 0.30258527 0.9089766 0.7702627 0.7624301 0.3827980 0.8391236

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [29]:
df2 <- df1 %>% 
  gather(Key,Value,V1:V20)
head(df2)

Key Value    
1 V1  0.0962233
2 V1  0.2503401
3 V2  0.5581773
4 V2  0.1780675
5 V3  0.4389170
6 V3  0.1040479

## Assignment 2

We will generate some data:

In [30]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [31]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [32]:
df5 <- full_join(df3,df4)
df5

Joining, by = "id"


id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook